.github/workflows/test-action.yml

```diff
@@ -1,427 +0,0 @@
-name: Testing
-
-on:
-  workflow_call:
-    inputs:
-      changed-files:
-        required: true
-        type: string
-        description: JSON string containing information about changed files
-
-jobs:
-  Linux:
-    runs-on: ubuntu-latest
-    timeout-minutes: 30
-
-    strategy:
-      fail-fast: false
-      max-parallel: 5
-      matrix:
-        python-version:
-          - '3.8'
-          - '3.9'
-        salt-version:
-          - '3005.4'
-          - '3006.4'
-        include:
-          - python-version: '3.10'
-            salt-version: '3006.4'
-
-    steps:
-    - uses: actions/checkout@v4
-
-    - uses: eLco/setup-vault@v1.0.2
-      with:
-        vault_version: 1.15.4
-
```

Comment:
**question (blocking)**: How can we make sure this runs as part of the workflow when using central artifacts? Without Vault installed, most of the test suite will be skipped.

Reply:
For now, I've added a flag for setting up Vault before the tests run. We'll roll with that for a while and see if we get too much sprawl from other extensions requirements.